velox/exec/Window.cpp

```diff
@@ -187,6 +193,31 @@ void Window::createWindowFunctions() {
   }
 }
 
+// The supportRowLevelStreaming is designed to support 'rank' and
+// 'row_number' functions and the agg window function with default frame.
+bool Window::supportRowLevelStreaming() {
+  for (const auto& windowNodeFunction : windowNode_->windowFunctions()) {
+    const auto& functionName = windowNodeFunction.functionCall->name();
+    auto windowFunctionEntry =
+        exec::getWindowFunctionEntry(functionName).value();
+    if (windowFunctionEntry->processingUnit == ProcessingUnit::kPartition) {
+      return false;
+    }
+
+    const auto& frame = windowNodeFunction.frame;
+    bool isDefaultFrame =
+        (frame.startType == core::WindowNode::BoundType::kUnboundedPreceding &&
+         frame.endType == core::WindowNode::BoundType::kCurrentRow);
+    // Only support the agg window function with default frame.
+    if (!(functionName == "rank" || functionName == "row_number") &&
+        !isDefaultFrame) {
```

Comment:
Is aggregate function covered here?
Maybe, this is not our expected checking logic. I guess if it's not the DefaultFrame, we should return false regardless of the function name. Right?

Reply:
@PHILO-HE For functions such as rank or row_number, the frame type is not considered. Only aggregate window functions take into account whether the frame is the default.